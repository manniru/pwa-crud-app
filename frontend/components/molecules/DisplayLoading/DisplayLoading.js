// import * as sc from './DisplayLoading.style';

const DisplayLoading = () => <div>Loading...</div>;

export default DisplayLoading;
