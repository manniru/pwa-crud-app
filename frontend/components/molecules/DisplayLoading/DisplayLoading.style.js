// import styled from 'styled-components';

// import { ReactComponent as loadingIcon } from '../../static/loading.svg';

// const spin = keyframes`
//   to {
//     transform: rotate(360deg);
//   }
// `;

// const divLoading = styled.p`
//   display: block;
//   margin: auto;
//   path: {
//     transform-origin: center;
//   }
// `;
