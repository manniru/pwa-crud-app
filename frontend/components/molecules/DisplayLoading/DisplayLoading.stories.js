import DisplayLoading from './DisplayLoading';

export default { title: 'Components|Molecules', component: DisplayLoading };

export const displayLoading = () => <DisplayLoading />;
