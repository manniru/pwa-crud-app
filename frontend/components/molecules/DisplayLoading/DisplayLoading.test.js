import { render, cleanup, prettyDOM, fireEvent } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import DisplayLoading from './DisplayLoading';
import theme from '../../../public/styles/theme.style';

const setup = (updatedProps = {}) => {
  const initialProps = {};
  const mergedProps = { ...initialProps, ...updatedProps };

  const result = render(
    <ThemeProvider theme={theme}>
      <DisplayLoading {...mergedProps} />
    </ThemeProvider>
  );

  const displayLoading = () => result.queryByText('Loading...');

  return { ...result, displayLoading };
};

describe('DisplayLoading', () => {
  afterEach(cleanup);

  it('matches snapshot', () => {
    const utils = setup();

    expect(utils.baseElement).toMatchSnapshot();
  });

  it('renders elements', () => {
    const utils = setup();

    expect(utils.displayLoading()).toBeInTheDocument();
  });
});
