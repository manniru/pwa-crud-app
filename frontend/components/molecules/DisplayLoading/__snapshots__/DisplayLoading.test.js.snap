// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayLoading matches snapshot 1`] = `
<body>
  <div>
    <div>
      Loading...
    </div>
  </div>
</body>
`;
